### PR TITLE
feat: move blame feature to an extension

### PR DESCRIPTION
**Pull request by sqs**
The blame feature (that shows the Git blame info for the selected line) is moving to the [Git extras extension](https://github.com/sourcegraph/sourcegraph-git-extras). The extension API perfectly supports this kind of feature.

Depends on https://github.com/sourcegraph/sourcegraph/pull/564.

screenshot from 2018-10-28 17-55-20
screenshot from 2018-10-28 17-55-02

> This PR updates the CHANGELOG.md file to describe any user-facing changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ All notable changes to Sourcegraph are documented in this file.
 
 ### Changed
 
+- The Git blame information shown at the end of a line is now provided by the [Git extras extension](https://sourcegraph.com/extensions/sourcegraph/git-extras). You must add that extension to continue using this feature.
```
Comment (slimsag) [inline CHANGELOG.md]: Why don't we turn this on by default? It seems like a feature regression otherwise, to me.

Comment (sqs) [reply CHANGELOG.md]: Yeah, I need to figure out

- how do we enable extensions by default on new instances (and existing instances when they upgrade)
- when we enable this in global settings (for all users) on Sourcegraph.com, how do we avoid the terrible UX of unauthed users clicking blame on GitHub and getting an error

I am going to figure out these things before merging this PR.

**Comment by sqs** [comment]
> I think you should remove the View commit prefix

Done :+1: 

> The lack of spacing here really sucks / looks ugly :/ Code discussions also uses a space prefixing hack (this only works because we highlight the entire line, not just the after line portion). Let's file an issue for tracking this.

Agreed, filed https://github.com/sourcegraph/sourcegraph/issues/567.

> It would be much nicer if this only showed blame on selected lines, let's file an issue both for tracking this + supporting this from an extension API POV.

Agreed. Incidentally, we have gotten some user/customer requests to see blame on all lines (not just the selected line), and that is better for certain use cases, so we should keep the current behavior where you *can* explicitly toggle it on for all lines.

**Comment by sqs** [comment]
Now that we have an [Explore area](https://github.com/sourcegraph/sourcegraph/pull/594) with basic extension recommendations, I'm going to merge this:

- I will make it NOT show the `Blame` button on GitHub because GitHub already has a `Blame` button. Users can toggle it on in the command palette. (TODO: need to make this possible in the extension API.)
- I will enable it in global settings on Sourcegraph.com so all users have it.

**Pull request merged**